client/cli/src/commands/vanity.rs

```diff
@@ -116,21 +116,21 @@ where
 
 fn good_waypoint(done: u64) -> u64 {
 	match done {
-		0..=1_000_000 => 100_000,
-		0..=10_000_000 => 1_000_000,
-		0..=100_000_000 => 10_000_000,
+		n if n <= 1_000_000 => 100_000,
```

Comment:
It was the [overlapping match](https://rust-lang.github.io/rust-clippy/master/index.html#match_overlapping_arm) lint. The code is correct.
I changed it if we want to enable the lint later to avoid pitfalls when using less obvious ranges (with open and closed intervals...).

Reply:
Have switched it to just not overlap the ranges. It's a little more visual clutter but it's fairly clear what's going on.